examples-copier/services/github_write_to_source_test.go

```diff
@@ -0,0 +1,135 @@
+	// Set up files to deprecate
+	originalFiles := FilesToDeprecate
+	defer func() {
+		FilesToDeprecate = originalFiles
+	}()
+
+	FilesToDeprecate = map[string]Configs{
+		"examples/old-example.go": {
+			TargetRepo:   "test/target",
+			TargetBranch: "main",
+		},
+		"examples/deprecated.go": {
+			TargetRepo:   "test/target",
+			TargetBranch: "main",
+		},
+	}
+
+	// Note: This test will fail if it actually tries to call GitHub API
+	// In a real test environment, we would need to:
+	// 1. Mock the GetRestClient() function
+	// 2. Mock the GitHub API responses
+	// 3. Verify the correct API calls were made
+	//
+	// For now, this test documents the expected behavior
+	// The actual implementation would require refactoring to inject dependencies
+
+	// Since we can't easily test this without mocking, we'll skip the actual call
+	t.Skip("Skipping test that requires GitHub API mocking")
+}
+
+func TestFilesToDeprecate_GlobalVariable(t *testing.T) {
+	// Test that we can manipulate the global FilesToDeprecate variable
+	originalFiles := FilesToDeprecate
+	defer func() {
+		FilesToDeprecate = originalFiles
+	}()
+
+	// Set test files (FilesToDeprecate is a map[string]Configs)
+	testFiles := map[string]Configs{
+		"file1.go": {TargetRepo: "test/repo1", TargetBranch: "main"},
+		"file2.go": {TargetRepo: "test/repo2", TargetBranch: "develop"},
+		"file3.go": {TargetRepo: "test/repo3", TargetBranch: "main"},
+	}
+	FilesToDeprecate = testFiles
+
+	if len(FilesToDeprecate) != 3 {
+		t.Errorf("FilesToDeprecate length = %d, want 3", len(FilesToDeprecate))
+	}
+
+	for file, config := range testFiles {
+		if deprecatedConfig, exists := FilesToDeprecate[file]; !exists {
+			t.Errorf("FilesToDeprecate missing file %s", file)
+		} else if deprecatedConfig.TargetRepo != config.TargetRepo {
+			t.Errorf("FilesToDeprecate[%s].TargetRepo = %s, want %s", file, deprecatedConfig.TargetRepo, config.TargetRepo)
+		}
+	}
+}
+
+func TestDeprecationFileEnvironmentVariables(t *testing.T) {
+	// Test that deprecation file configuration can be set via environment variables
+	// The UpdateDeprecationFile function uses os.Getenv to read these values
+
+	tests := []struct {
+		name              string
+		deprecationFile   string
+	}{
+		{
+			name:              "default config",
+			deprecationFile:   "deprecated-files.json",
+		},
+		{
+			name:              "custom file",
+			deprecationFile:   "custom-deprecated.json",
+		},
+		{
+			name:              "nested path",
+			deprecationFile:   "docs/deprecated/files.json",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// The deprecation file path is typically configured via environment variables
+			// This test documents the expected configuration approach
+			if tt.deprecationFile == "" {
+				t.Error("Deprecation file path should not be empty")
+			}
+		})
+	}
+}
+
+// Note: Comprehensive testing of UpdateDeprecationFile would require:
```

Comment:
Are we leaving this here as a thing we eventually want to do? I'm wondering if we do/should have a Jira ticket to capture this as future work?

Reply:
good idea. i'd like to get this done eventually, so i'll change to a TODO with accompanying Jira ticket (here and the other instances)